src/libraries/System.Private.CoreLib/src/System/IO/BufferedFileStreamStrategy.cs

```diff
@@ -0,0 +1,132 @@
+            }
+        }
+
+        internal override bool IsClosed => _strategy.IsClosed;
+
+        internal override void Lock(long position, long length) => _strategy.Lock(position, length);
+
+        internal override void Unlock(long position, long length) => _strategy.Unlock(position, length);
+
+        public override long Seek(long offset, SeekOrigin origin) => _bufferedStream.Seek(offset, origin);
+
+        public override void SetLength(long value) => _bufferedStream.SetLength(value);
+
+        public override int ReadByte() => _bufferedStream.ReadByte();
+
+        public override IAsyncResult BeginRead(byte[] buffer, int offset, int count, AsyncCallback? callback, object? state)
+            => _bufferedStream.BeginRead(buffer, offset, count, callback, state);
+
+        public override int EndRead(IAsyncResult asyncResult)
+            => _bufferedStream.EndRead(asyncResult);
+
+        public override int Read(byte[] buffer, int offset, int count) => _bufferedStream.Read(buffer, offset, count);
+
+        public override int Read(Span<byte> buffer) => _bufferedStream.Read(buffer);
+
+        public override Task<int> ReadAsync(byte[] buffer, int offset, int count, CancellationToken cancellationToken)
+            => _bufferedStream.ReadAsync(buffer, offset, count, cancellationToken);
+
+        public override ValueTask<int> ReadAsync(Memory<byte> buffer, CancellationToken cancellationToken = default)
+            => _bufferedStream.ReadAsync(buffer, cancellationToken);
+
+        public override IAsyncResult BeginWrite(byte[] buffer, int offset, int count, AsyncCallback? callback, object? state)
+            => _bufferedStream.BeginWrite(buffer, offset, count, callback, state);
+
+        public override void EndWrite(IAsyncResult asyncResult)
+            => _bufferedStream.EndWrite(asyncResult);
+
+        public override void WriteByte(byte value) => _bufferedStream.WriteByte(value);
+
+        public override void Write(byte[] buffer, int offset, int count) => _bufferedStream.Write(buffer, offset, count);
+
+        public override void Write(ReadOnlySpan<byte> buffer) => _bufferedStream.Write(buffer);
+
+        public override Task WriteAsync(byte[] buffer, int offset, int count, CancellationToken cancellationToken)
+            => _bufferedStream.WriteAsync(buffer, offset, count, cancellationToken);
+
+        public override ValueTask WriteAsync(ReadOnlyMemory<byte> buffer, CancellationToken cancellationToken = default)
+            => _bufferedStream.WriteAsync(buffer, cancellationToken);
+
+        public override void Flush() => _bufferedStream.Flush();
+
+        internal override void Flush(bool flushToDisk) => _bufferedStream.Flush(flushToDisk);
+
+        public override Task FlushAsync(CancellationToken cancellationToken)
+            =>  _bufferedStream.FlushAsync(cancellationToken);
+
+        public override Task CopyToAsync(Stream destination, int bufferSize, CancellationToken cancellationToken)
```

Comment:
We don't need / want to also override CopyTo?

Reply:
@stephentoub that's an excellent catch! I am going to add this asap